### PR TITLE
fix(cjs): accept dotted canonical command form (#3243)

### LINKED ISSUE #3243: workflow files use dot notation but CLI only accepts space-separated subcommand
### GSD Version

GSD version: v1.41.0 (latest stable) - original install came from npm

### Runtime

Claude Code

### Operating System

WSL

### Node.js Version

Node: v22.22.2 

### Shell

bash

### Installation Method

Updated from a previous version

### What happened?

  Workflow markdown files emit gsd-sdk query calls using dot notation (e.g.
  roadmap.get-phase), but the SDK CLI router only accepts space-separated     
  subcommands (e.g. roadmap get-phase). This causes Error: Unknown command
  failures at runtime.                                                        
                                                                        
  This installation was originally set up via NPM and later migrated to the   
  GitHub-based installer. It's possible the dot notation worked under the old
  NPM distribution and broke during that transition, or that stale artifacts  
  from the NPM install are involved.                                    
        
  command-aliases.generated.cjs lists dot notation as canonical and           
  space-separated as an alias, but the CLI router does not resolve the
  canonical dot form.                                                         
                                                                        
  Additionally, 14 commands used in workflows have no entry in                
  command-aliases.generated.cjs at all: check.decision-coverage-plan,
  check.decision-coverage-verify, frontmatter.get, frontmatter.set,           
  learnings.copy, milestone.complete, phase.mvp-mode, progress.bar,     
  requirements.mark-complete, stats.json, task.is-behavior-adding,
  todo.match-phase, uat.render-checkpoint, workstream.list.

### What did you expect?

  gsd-sdk query roadmap.get-phase "21.1" should work the same as gsd-sdk query
   roadmap get-phase "21.1", since the workflows use dot notation throughout. 
   

### Steps to reproduce

  1. Run `gsd-sdk query roadmap.get-phase "21.1"`
  2. Observe: Error: Unknown command: roadmap.get-phase                       
  3. Run `gsd-sdk query roadmap get-phase "21.1"`
  4. Observe: works correctly, returns JSON                                   
       

### Error output / logs

```shell
$ gsd-sdk query roadmap.get-phase "21.1"
  Error: Unknown command: roadmap.get-phase
```

### GSD Configuration

```json

```

### GSD State (if relevant)

```markdown

```

### Runtime settings.json (if relevant)

```json

```

### How often does this happen?

Every time (100% reproducible)

### Impact

Blocker — Cannot use GSD at all

### Workaround (if any)

  Use space-separated format instead of dot notation: gsd-sdk query roadmap 
  get-phase instead of gsd-sdk query roadmap.get-phase. Affects ~54 workflow  
  files including plan-phase.md, execute-phase.md, progress.md, and
  verify-work.md.    

### Additional context

installation originally NPM, now GitHub installer

### Privacy Checklist

- [x] I have reviewed all pasted output for PII (usernames, paths, API keys) and redacted where necessary

### PR DESCRIPTION
**Pull request by trek-e**
## Fix PR

---

## Linked Issue

Fixes #3243

---

## What was broken

`gsd-tools <domain>.<subcommand>` (dotted canonical form, e.g. `state.update`, `roadmap.get-phase`) failed with `Error: Unknown command: <domain>.<subcommand>`. The CJS dispatcher's `switch (command)` only accepted the spaced form (`state update`, `roadmap get-phase`). The dotted form was only split client-side by the SDK's `dottedCommandToCjsArgv` helper — any caller that bypassed the SDK hit the error unconditionally.

The reporter's specific symptom is most likely a stale npm-installed `get-shit-done-cc` binary shadowing the modern GitHub-installer binary (the old binary lacks the SDK split). Uninstalling the stale binary (`npm uninstall -g get-shit-done-cc`) would have resolved the immediate symptom — but this PR also hardens the CJS dispatcher itself so that any future caller (stale binary, workflow shell-out, third-party script) can use the canonical dotted form natively.

## What this fix does

Adds a shim at the top of `main()` in `gsd-tools.cjs` that splits `args[0]` on the first `.` when present and normalizes it to the spaced form before the `switch` statement is reached. `generate-slug.hello-world` becomes `command='generate-slug'`, `args=['generate-slug','hello-world']`. The shim guards against empty head/rest so `.hidden` and bare `.` args fall through unchanged to the existing "Unknown command" path.

Also improves the "Unknown command" error message to suggest the spaced equivalent when a dotted form was passed: `Unknown command: foo — did you mean: "foo bar"?`

## Root cause

The `switch (command)` in `runCommand` in `gsd-tools.cjs` only matched top-level command names (`'state'`, `'roadmap'`, etc.). The dotted form (`'state.update'`) was only ever normalized client-side by the SDK. Any caller bypassing the SDK received `Unknown command: state.update` from the CJS dispatcher's `default:` branch.

### How I verified the fix

- `runGsdTools(['generate-slug.hello-world'])` now returns the same slug as `runGsdTools(['generate-slug', 'hello-world'])`.
- `runGsdTools(['current-timestamp.date'])` returns the same output as the spaced form.
- `runGsdTools(['validate.plan'])`, `runGsdTools(['roadmap.analyze'])`, `runGsdTools(['phases.list'])` no longer emit "Unknown command: <dotted>".
- `runGsdTools(['check.decision-coverage-plan'])` (multi-dot) splits on the first dot — the error (if any) is about `check`, not the verbatim dotted form.
- `runGsdTools(['foo.bar'])` error message now includes a suggestion: `did you mean: "foo bar"?`
- Leading-dot args like `['.hidden']` are still rejected correctly.

### Regression test added?

- [x] Yes — added a test that would have caught this bug

`tests/bug-3243-dotted-command-form.test.cjs` — 9 tests covering: dotted form produces identical output to spaced form, commands route to their handlers (not "Unknown command"), multi-dot splits on first dot only, existing spaced-only callers are unaffected, leading-dot guard, and "Unknown command" suggestion for genuinely unknown dotted commands.

### Platforms tested

- [x] macOS
- [ ] Windows (including backslash path handling)
- [ ] Linux
- [x] N/A (not platform-specific)

### Runtimes tested

- [x] Claude Code
- [ ] Gemini CLI
- [ ] OpenCode
- [ ] Other: ___
- [ ] N/A (not runtime-specific)

---

## Checklist

- [x] Issue linked above with `Fixes #NNN` — **PR will be auto-closed if missing**
- [x] Linked issue has the `confirmed-bug` label
- [x] Fix is scoped to the reported bug — no unrelated changes included
- [x] Regression test added (or explained why not)
- [x] All existing tests pass (`npm test`)
- [ ] `.changeset/` fragment added if this is a user-facing fix (`npm run changeset -- --type Fixed --pr <NNN> --body "..."`) — or `no-changelog` label applied
- [x] No unnecessary dependencies added

## Breaking changes

None — additive only. The spaced form continues to work exactly as before. The dotted form now works in addition to the spaced form.

---

## Notes on the 14 missing manifest entries

The issue also reports 14 commands with no entry in `command-aliases.generated.cjs` (`check.decision-coverage-plan`, `check.decision-coverage-verify`, `frontmatter.get`, `frontmatter.set`, `learnings.copy`, `milestone.complete`, `phase.mvp-mode`, `progress.bar`, `requirements.mark-complete`, `stats.json`, `task.is-behavior-adding`, `todo.match-phase`, `uat.render-checkpoint`, `workstream.list`). This PR does not address those — extending the alias manifest requires locating or creating manifest source files in `sdk/src/query/command-manifest.*.ts`, which is out of scope for this bug fix. A follow-on issue should be filed to track that work separately.

## Summary by CodeRabbit

* **Bug Fixes**
  * CLI now accepts dotted command syntax (e.g., `state.update`) as equivalent to space-separated form, preventing spurious "Unknown command" errors and providing a helpful spaced-form suggestion when appropriate.

* **Tests**
  * Added regression tests verifying dotted vs. spaced command equivalence, multi-dot behavior, leading-dot and spaced-edge cases, and correct error/suggestion messages.

### CODE REVIEW

**Comment by trek-e** [comment]
@coderabbitai full review

**Comment by trek-e** [comment]
@coderabbitai full review

**Comment by trek-e** [comment]
@coderabbitai full review

File: tests/bug-3243-dotted-command-form.test.cjs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    // (the dotted form); it should say something about "check" (the split result).
+    const errText = dotted.error || '';
+    assert.ok(
+      !errText.includes('Unknown command: check.decision-coverage-plan'),
+      [
+        'multi-dot dotted form must not be passed verbatim to "Unknown command".',
+        'Got error:', errText,
+      ].join('\n')
+    );
+  });
```
[the remark this answers is not in the document]
Comment (trek-e) [reply tests/bug-3243-dotted-command-form.test.cjs:126]: Out of scope for #3243 (which just adds the dotted-command shim). The `--json-errors` mode suggested here does not currently exist in `gsd-tools.cjs` — adding it is a meaningful new feature that would be a broader change than this PR warrants. Filed as #3255 for separate follow-on work.

One clarification worth noting: the current `result.error.includes('foo bar')` assertions do **not** violate the repo's `lint-no-source-grep` rule — that rule forbids grepping *source file content* (e.g. `readFileSync` + `.includes()`). These tests assert on *runtime CLI output*, which is a different and legitimate pattern. The structured-error surface is still worth building (#3255), but the existing tests are correct and durable for now.
